modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ActionTask.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal;
+
+/**
+ * An action task to be executed.
+ */
+public interface ActionTask {
```

Comment:
In which sense this is different than a Runnable or any other java interface done for that? Use that instead of defining a different interface

Reply:
The difference is that I need to return a boolean after executing the action. run() from Runnable returns a void, do you know any other I can use?

Reply:
You can use a Supplier<Boolean>